rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/CovidRDTBarcodeFactory.java

```diff
@@ -33,14 +34,18 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         final JsonApi jsonApi = (JsonApi) widgetArgs.getContext();
         if (requestCode == BARCODE_REQUEST_CODE && resultCode == RESULT_OK && data != null) {
             try {
-                String barcodeVals = getBarcodeValsAsCSV(data);
-                jsonApi.writeValue(widgetArgs.getStepName(),
-                        widgetArgs.getJsonObject().optString(JsonFormConstants.KEY),
-                        barcodeVals, "", "", "", false);
+                if (data.getBooleanExtra(CovidConstants.IntentKeys.BARCODE_SENSOR_TRIGGER, false)) {
+                    sensorTriggered();
+                } else {
+                    String barcodeVals = getBarcodeValsAsCSV(data);
+                    jsonApi.writeValue(widgetArgs.getStepName(),
+                            widgetArgs.getJsonObject().optString(JsonFormConstants.KEY),
+                            barcodeVals, "", "", "", false);
 
-                String[] individualVals = splitCSV(barcodeVals);
-                populateRelevantFields(individualVals);
-                moveToNextStep(convertDate(individualVals[1], "YYYY-MM-dd"));
+                    String[] individualVals = splitCSV(barcodeVals);
+                    populateRelevantFields(individualVals);
+                    moveToNextStep(convertDate(individualVals[1], "YYYY-MM-dd"));
+                }
```

Comment:
Even when the sensor is triggered, all other steps, like populating relevant fields, should be performed.

I would recommend modifying `moveToNextStep` to add the sensor trigger check.

Reply:
fixed.